src/Constraint/ComparisonDumper.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver\Constraint;
+
+use Composer\Semver\VersionParser;
+
+class ComparisonDumper
+{
+    /**
+     * @var VersionParser
+     */
+    private $versionParser;
+
+    /**
+     * @param VersionParser $versionParser
+     */
+    public function __construct(VersionParser $versionParser = null)
+    {
+        if (null === $versionParser) {
+            $versionParser = new VersionParser();
+        }
+
+        $this->versionParser = $versionParser;
+    }
+
+    /**
+     * Takes a given constraint and a given version and creates
+     * the version_compare() statements required to evaluate
+     * whether a version matches upper and lower bounds.
+     *
+     * @param ConstraintInterface $constraint
+     * @param string              $version
```

Comment:
I don't understand why you are passing a constraint and a version? Shouldn't you only be passing a constraint to this?

Reply:
Depends. In case of the planned PHP version check we have access to this library and can generate the complete string right away. In case you only want to dump is and add the version later, we‘d need `%s` so you can sprintf(), I guess.

Reply:
We don't have the runtime version when composer runs as that might be different php/env. That's the whole point of the feature.

Reply:
So just a placeholder then, right?

Reply:
Okay, implemented in 5bb8bc8. We do have to normalize the versions at runtime which is why the generated code needs quite a bit of logic but I really kept it down to the bare minimum:

* Only the runtime version is normalized at runtime, the bounds are when dumping the comparison.
* Only what's really needed is dumped, no variables for upper or lower bounds if they are lowermost or uppermost.

Using `eval()` here. We could also expect to just set `$version` before the dump, so we don't have to use `eval()`.

Reply:
eval is fine in the test IMO. It should end up being dumped in a file that gets executed as usual anyway. What I'd like tho is that %version% would not be wrapped in quotes in the dumped code, that way it's easy to replace it with smth like PHP_VERSION or whatever is needed.

Reply:
Right, done in 108aad5d81300af55c305a89d4ece459beb52c44.